src/crypto/SymmetricCipher.cpp

```diff
@@ -197,6 +197,8 @@ QString SymmetricCipher::modeToString(const Mode mode)
         return QStringLiteral("AES-128/CBC");
     case Aes256_CBC:
         return QStringLiteral("AES-256/CBC");
+    case Aes256_CBC_UNPADDED:
```

Comment:
The AES256-CBC implementation here is encrypting exactly-32-byte-long key parts which are randomly generated. So padding doesn't really make sense.

Reply:
I added this because decryption did actually fail with aligned blocks. Botan expects there to be a minimum of one padding block - how else would it know how much padding had been added?

In other words, if you have a 31 byte block, that would have one byte of padding and end up 32 bytes long. But if you have a 32 byte long block... How do you know on decryption whether it's 32 or 31 bytes before padding? There must be at least one byte of padding to store the padding length itself...

I think we need this "new" algorithm (well, variant).

Reply:
Hmmmm good point